metadata-io/src/main/java/com/linkedin/metadata/entity/AspectUtils.java

```diff
@@ -73,9 +73,10 @@ private static MetadataChangeProposal getProposalFromAspect(String aspectName, R
     try {
       MetadataChangeProposal proposal = original.copy();
       GenericAspect genericAspect = GenericRecordUtils.serializeAspect(aspect);
-      // Set UPSERT changetype here as additional changes being added should always be
-      // done in UPSERT mode even for patches
-      // proposal.setChangeType(ChangeType.UPSERT);
+      // Additional changes should never be set as PATCH, if a PATCH is coming across it should be an UPSERT
+      if (ChangeType.PATCH.equals(proposal.getChangeType())) {
```

Comment:
Looks good.

Reply:
Do we have unit tests for this class?

Reply:
Not directly, but it does get used on the OpenAPI controllers so it does get executed in tests